src/__tests__/types.tst.ts

```diff
@@ -1,7 +1,7 @@
 import { describe, expect, it } from "tstyche"
-import type { GetCommonVariableForEnv, GetName } from "../types"
-import { createEnvironmentRegistry } from "../environment-registry"
-import { defineType } from "../type-def"
+import type { GetCommonVariableForEnv, GetName } from "../types.js"
+import { createEnvironmentRegistry } from "../environment-registry.js"
+import { defineType } from "../type-def.js"
```

Comment:
With the `exports` field set in `package.json`, you can also import from `envoyage`:

```suggestion
import { defineType } from "envoyage"
```

This would make TypeScript look at `package.json` and resolve `defineType` similar how it would be resolved on user side. The type test would ensure that `defineType` is actually exported from the package.